src/pages/en/reference/integrations-reference.md

```diff
@@ -40,30 +39,159 @@ interface AstroIntegration {
+
+The **`stage`** denotes how this script (the `content`) should be inserted. Some stages are inserted purely, while others are picked up by Vite:
+
+- `"before-hydration"`: Imported client-side, before the hydration script runs. Processed & resolved by Vite.
+- `"head-inline"`: Injected into a script tag in the `<head>` of every page. Not processed or resolved by Vite.
+- `"page"`: Injected into the JavaScript bundle of every page. Processed & resolved by Vite.
+- `"page-ssr"`: Injected into the frontmatter of every Astro page. Processed & resolved by Vite.
+
 ### astro:config:done
+
+**When it's run:** after the Astro config has resolved and other integrations have run their `astro:config:setup` hooks.
+
+**Use case:** to retrieve the final config for use in other hooks.
+
+#### "config" option
+
+**Type**: `AstroConfig`
+
+A read-only copy of the user-supplied [Astro config](/en/reference/configuration-reference/). This is resolved _after_ other integrations have run.
+
 ### astro:server:setup
+
+**When it's run:** just after the Vite server is created in "dev" or "preview" mode, but before the `listen()` event is fired. [See Vite's createServer API](https://vitejs.dev/guide/api-javascript.html#createserver) for more.
+
+**Use case:** to update Vite server options and middleware.
+
+#### "server" option
+
+**Type:** [`ViteDevServer`](https://vitejs.dev/guide/api-javascript.html#vitedevserver)
+
+An mutate-able instance of the Vite server used in "dev" and "preview" mode. For instance, this is [used by our Partytown integration](https://github.com/withastro/astro/tree/main/packages/integrations/partytown) to inject the Partytown server as middleware:
+
+```js
+'astro:server:setup': ({ server }) => {
+  server.middlewares.use(
+    sirv(partytownLibDirectory, {
+      mount: '/~partytown',
+      ...
+    })
+  );
+}
+```
+
 ### astro:server:start
+
+**When it's run:** after the server's `listen()` event has fired.
+
+**Use case:** to intercept network requests at the specified address. If you intend to use this address for middleware, consider using `astro:server:setup` instead.
+
+#### "address" option
+
+**Type:** [`AddressInfo`](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules__types_node_net_d_._net_.addressinfo.html)
+
+The address, family and port number supplied by the [NodeJS Net module](https://nodejs.org/api/net.html).
+
 ### astro:server:done
+
+**When it's run:** when the dev server is closed.
+
+**Use case:** to run any cleanup events you may trigger during the `astro:server:setup` or `astro:server:start` hooks.
+
 ### astro:build:start
+
+**When it's run:** after the `astro:config:done` event, but before the production build begins.
+
+**Use case:** to setup any global objects or clients you may need during a production build. This can also extend build configuration options in the [experimental adapter API](/en/reference/adapter-reference/).
```

Comment:
```suggestion
**Use case:** To setup global objects or clients which maybe needed during a production build. This can also extend the build configuration options in the [experimental adapter API](/en/reference/adapter-reference/).
```